rest/src/main/kotlin/builder/interaction/PublicInteractionBuilder.kt

```diff
@@ -58,6 +66,16 @@ class PublicInteractionResponseCreateBuilder :
         embeds!! += EmbedBuilder().apply(builder)
     }
 
+    @OptIn(ExperimentalContracts::class)
+    @KordPreview
+    inline fun components(builder: ActionRowBuilder.() -> Unit) {
+        contract {
+            callsInPlace(builder, InvocationKind.EXACTLY_ONCE)
+        }
+
+        components = mutableListOf(ActionRowBuilder().apply(builder).build())
+    }
```

Comment:
* can messages contain buttons without action rows? If so, this is a misnomer. If not, the `components` field should not be able to accept buttons as it currently does.

* this DSL does not allow you to add more than one component, while messages can contain up to 5 action rows.

Reply:
> can messages contain buttons without action rows?
No

> If so, this is a misnomer.

Why it allows you to add components even though you can only add one

> If not, the components field should not be able to accept buttons as it currently does.

How does it accept buttons? it only accepts an action row

Reply:
> Why it allows you to add components even though you can only add one
> How does it accept buttons? it only accepts an action row

Not this DSL, but the builder's `components` accepts built `DiscordComponent`, which is also what buttons build to.